tests/phpunit/integration/Core/User_Input/User_InputTest.php

```diff
@@ -197,4 +207,64 @@ public function test_set_answers() {
 			$response
 		);
 	}
+
+	public function test_set_answers__preserves_purpose_answer_user_attribution() {
+		$second_admin_id = $this->factory()->user->create( array( 'role' => 'administrator' ) );
+
+		$this->user_input->set_answers(
+			array(
+				'purpose'       => array( 'publish_blog' ),
+				'postFrequency' => array( 'daily' ),
+				'goals'         => array( 'growing_audience', 'finding_new_topics' ),
+			)
+		);
+
+		$existing_answers = $this->user_input->get_answers();
+
+		$this->assertEquals( $existing_answers['purpose']['answeredBy'], $this->user_id );
+
+		$this->user_options->switch_user( $second_admin_id );
+
+		$this->user_input->set_answers(
+			array(
+				'purpose'       => array( 'publish_blog' ),
+				'postFrequency' => array( 'weekly' ),
+				'goals'         => array( 'improving_performance' ),
+			)
+		);
+
+		$existing_answers = $this->user_input->get_answers();
+		// Since purpose answer didn't change, it should still be attributed to admin 1.
+		$this->assertEquals( $existing_answers['purpose']['answeredBy'], $this->user_id );
```

Comment:
Maybe we should check that some changed here, and have one more test for what happens when _no_ answers change, just to cover all scenarios.